application/comit_node/src/http_api/rfc003/swap.rs

```diff
@@ -339,7 +339,10 @@ pub fn get_swaps<T: MetadataStore<SwapId>, S: StateStore<SwapId>>(
             response.with_resources("swaps", swaps);
             Ok(warp::reply::json(&response))
         }
-        Err(e) => Err(warp::reject::custom(HttpApiProblemStdError::new(e))),
+        Err(e) => {
+            error!("returning error: {:?}", e);
```

Comment:
❌ Log statements like this in controllers are suboptimal IMO:

1. If it's a 400 type error we may not want to log it as `error`. `error` should be for actual problems with the system (not invalid input from users).
2. If we need to log these kind of errors it will only catch this one route (adding a warp filter to log every 500 error on all routes would catch everything and remove this stuff from the controller).
3. If it is a 500 type error we probably want to log it before it's turned into a `HttpApiProblem`, so closer to the source of the error.

My preferred resolution would be to just remove this for now and have a spike to come up with a sustainable approach to logging. @bonomat @luckysori thoughts?

Reply:
The issue is that without this log it is a pain to investigate when the error appears as there is no error logged.
The only way to investigate is to re-add the error here and re-run the test hoping that the same error appears.
When it's intermittent, then it becomes a debug nightmare.

What about opening a spike and adding a `// FIXME` here?

Reply:
Isn't the error your logging exactly the same as what is returned over the http api? Would adding the warp filter that logs 500 errors not give you what you need?

Reply:
FIXME + issue created